### ISSUE TITLE
`Service.controlWebServer(enable: true)` has no effect if `--enable-vm-service` was not provided at launch

### ISSUE THREAD

**Issue by bkonyi** [flutter/flutter#132865]
From the original issue (https://github.com/dart-lang/sdk/issues/53262):

> Steps:
> 
> 1. Create counter application.
> 2. Add a test file to the folder `test` with the code below
> 3. Run the tests
> 4. Observe both tests fail
> 
> Copy the test to a dart project and see the test succeed.
> 
> <details>
>  <summary>The test code:</summary>
> 
> ```
> 
> import 'dart:developer';
> 
> import 'package:flutter_test/flutter_test.dart';
> 
> Future<Uri> _serviceUri() async {
>   Uri? uri = (await Service.getInfo()).serverWebSocketUri;
> 
>   if (uri != null) return uri;
> 
>   uri = (await Service.controlWebServer(enable: true)).serverWebSocketUri;
> 
>   const timeout = Duration(seconds: 5);
>   final stopwatch = Stopwatch()..start();
> 
>   while (uri == null) {
>     if (stopwatch.elapsed > timeout) {
>       throw StateError(
>         'Could not start VM service.',
>       );
>     }
>     await Future.delayed(const Duration(milliseconds: 1));
>     uri = (await Service.getInfo()).serverWebSocketUri;
>   }
> 
>   return uri;
> }
> 
> void main() {
>   test('Service can be started for flutter test.', () async {
>     Uri? uri;
> 
>     uri = await _serviceUri();
> 
>     print(uri);
>   });
> 
>   testWidgets('Service can be started for flutter widgets test.',
>       (WidgetTester tester) async {
>     Uri? uri;
> 
>     await tester.runAsync(() async => uri = await _serviceUri());
> 
>     print(uri);
>   });
> }
> 
> 
> ```
> 
> 
> </details>

Investigation findings:

> So I've done some digging, and I think this might be a tricky one to resolve.
> 
> First off, I've always assumed that invoking [`Service.controlWebServer`](https://api.dart.dev/stable/3.1.0/dart-developer/Service/controlWebServer.html) with `enable = true` actually started the VM service isolate if it wasn't already running. As it turns out, that's not actually the case, and the API only controls whether or not the VM service will accept HTTP / WebSocket requests.
> 
> Another important point to remember is that the creation of isolates is not controlled completely by the Dart VM, as the embedder is responsible for initializing much of an isolate's state. The service isolate is no exception, and the embedder has a say in how and if the service isolate is initialized.
> 
> In the standalone embedder (i.e., `dart`), the service isolate is _always_ present in configurations that support it (i.e., not `PRODUCT` mode). The presence of `--observe` or `--enable-vm-service` is used to set the value of `_autoStart`, causing the web server to start [here](https://github.com/dart-lang/sdk/blob/main/sdk/lib/_internal/vm/bin/vmservice_io.dart#L385). If the HTTP server is not automatically started, invoking `Service.controlWebServer(enable: true)` causes the HTTP server to be initialized, making the VM service accessible.
> 
> On the other hand, the Flutter engine takes a different approach. Instead of always initializing the service isolate and leaving the HTTP server disabled by default, [if `--enable-vm-service` isn't provided the service isolate simply isn't started](https://github.com/flutter/engine/blob/main/runtime/dart_isolate.cc#L768). This means that invoking `Service.controlWebServer(enable: true)` has no effect in Flutter or Flutter tests if `--enable-vm-service` wasn't provided at launch.
> 
> Unfortunately, I don't think resolving this issue will be as simple as always starting the service isolate in the Flutter engine as that will lead to performance regressions in benchmarks that don't explicitly pass `--enable-vm-service`.
> 
> We could potentially change `flutter_tester` to always start the service isolate in a similar fashion to the standalone VM. This would still impact memory usage (and likely startup times), but would only impact tests instead of all Flutter configurations that support the VM service.

cc @polina-c

**Comment by bkonyi** [comment]
@zanderso, do you have any thoughts on this? In particular, WDYT about configuring `flutter_tester` to always have the service isolate available?

**Comment by zanderso** [comment]
I believe there is a group of folks working on reducing `flutter_tester` test startup time, so I think it will be necessary to quantify any possible slowdown.

**Comment by polina-c** [comment]
Thanks for details.
It is ok if I can suggest to use flag. My problem is that both `flutter test` and `flutter run` do not recognize the flag:

```
flutter test --enable-vm-service
Could not find an option named "enable-vm-service".
```


```
polinach$ flutter run test/connect_test.dart -d flutter-tester  --enable-vm-service
Could not find an option named "enable-vm-service".
```

Did i misspell spmehing?

**Comment by bkonyi** [comment]
Yeah, unfortunately flutter_tools uses `enable-vmservice` as the flag (we should probably deprecate that and add `enable-vm-service` instead to be consistent with the Dart SDK):

```
--[no-]enable-vmservice                                      Enables the VM service without "--start-paused". This flag is intended for use with tests that will use "dart:developer" to interact with the VM service at runtime.
                                                             This flag is ignored if "--start-paused" or coverage are requested, as the VM service will be enabled in those cases regardless.
```
